thunderbird/processes/wps_generate_climos.py

```diff
@@ -0,0 +1,296 @@
+            ),
+            LiteralInput(
+                "split_vars",
+                "Split Variables",
+                default=True,
+                abstract="Generate a separate file for each dependent variable in the file",
+                data_type="boolean",
+            ),
+            LiteralInput(
+                "split_intervals",
+                "Split Intervals",
+                default=True,
+                abstract="Generate a separate file for each climatological period",
+                data_type="boolean",
+            ),
+            LiteralInput(
+                "dry_run",
+                "Dry Run",
+                abstract="Checks file to ensure compatible with process",
+                data_type="boolean",
+            ),
+        ]
+        outputs = [
+            ComplexOutput(
+                "output",
+                "Output",
+                abstract="Collection of climatoligical files",
+                as_reference=True,
+                supported_formats=[FORMATS.META4],
+            ),
+            ComplexOutput(
+                "dry_output",
+                "Dry Output",
+                as_reference=True,
+                abstract="File information",
+                supported_formats=[FORMATS.TEXT],
+            ),
+        ]
+
+        super(GenerateClimos, self).__init__(
+            self._handler,
+            identifier="generate_climos",
+            version="0.8.0",
+            title="Generate Climatological Means",
+            abstract="Generate files containing climatological means from input files of daily, monthly, or yearly data that adhere to the PCIC metadata standard (and consequently to CMIP5 and CF standards).",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def dry_run_info(self, filepath, climo):
+        output = ["Dry Run"]
+        output.append(f"File: {filepath}")
+        try:
+            input_file = CFDataset(filepath)
+        except Exception as e:
+            output.append(f"{e.__class__.__name__}: {e}")
+        else:
+            periods = input_file.climo_periods.keys() & climo
+            output.append(f"climo_periods: {periods}")
+            for attr in "project institution model emissions run".split():
+                try:
+                    output.append(f"{attr}: {getattr(input_file.metadata, attr)}")
+                except Exception as e:
+                    output.append(f"{attr}: {e.__class__.__name__}: {e}")
+            output.append(f"dependent_varnames: {input_file.dependent_varnames()}")
+            for attr in "time_resolution is_multi_year_mean".split():
+                output.append(f"{attr}: {getattr(input_file, attr)}")
+
+        filename = os.path.join(self.workdir, "dry.txt")
```

Comment:
Does every process get its own unique workdir? And do you *have* to use a named file and return it?

Is it a possibility to use `tempfile.TemporaryFile(suffix='.txt', dir=self.workdir)` or `tempfile.NamedTemporaryFile(...)`?

Reply:
Broader question: how and when do birds clean up their output?

Reply:
This is a good question.  From what I've seen so far it seems like each process call creates a directory in `/tmp/`.  It also doesn't clean up after the process is done (at least on my local machine tests), I don't know how it behaves when running it from a docker container.  I imagine the process results would go down with the container but I can't say for sure just yet.

I can try changing to a tempfile and see what happens.